include/exadg/aero_acoustic/calculators/source_term_calculator.h

```diff
@@ -153,6 +154,73 @@ class SourceTermCalculator
     }
   }
 
+  void
+  compute_source_term(dealii::MatrixFree<dim, Number> const &       matrix_free_in,
+                      VectorType &                                  dst,
+                      dealii::Function<dim> const &                 analytical_source_term,
+                      std::pair<unsigned int, unsigned int> const & cell_range) const
+  {
+    CellIntegratorScalar dpdt(matrix_free_in, data.dof_index_pressure, data.quad_index);
+
+    Number rho = static_cast<Number>(data.density);
+
+    auto apply_scaling = get_scaling_function();
+
+    for(unsigned int cell = cell_range.first; cell < cell_range.second; ++cell)
+    {
+      dpdt.reinit(cell);
+
+      for(unsigned int q = 0; q < dpdt.n_q_points; ++q)
+      {
+        scalar flux = -rho * FunctionEvaluator<0, dim, Number>::value(analytical_source_term,
+                                                                      dpdt.quadrature_point(q));
+
+        if(data.blend_in)
+          apply_scaling(flux, dpdt.quadrature_point(q));
+
+        dpdt.submit_value(flux, q);
+      }
+
+      dpdt.integrate_scatter(dealii::EvaluationFlags::values, dst);
+    }
+  }
+
+  std::function<void(scalar &, qpoint const &)>
+  get_scaling_function() const
+  {
+    // In case we blend in the source term, we check if the scaling is space dependent. Only in that
+    // case we have to evaluate the function in every equadrature point. Otherwise the scaling
+    // is purely temporal and constant during this function.
+    if(data.blend_in)
+    {
+      AssertThrow(data.blend_in_function != nullptr,
+                  dealii::ExcMessage("No blend-in function provided."));
+    }
+
+    if(data.blend_in_function)
+      data.blend_in_function->set_time(time);
```

Comment:
Is this code new? Is it correct that this code could be shifted to line 209/210 (`if(space_dependent_scaling)`)? Could you explain the reason for this change?

Reply:
I wanted to get rid of the member `time`, but decided it is better to set the time during setup of the scaling function and forgot to move `time` back to `value()`. [522b33b](https://github.com/exadg/exadg/pull/659/commits/522b33bf7e1f2d0831b10aad13c814bba7db8346)